Web Inspector: remove frontend and backend support for ApplicationCache

burg opened the pull request:
#### bb2b6735bb3443035edbee7ad712aa4c3a25fa5d
<pre>
Web Inspector: remove frontend and backend support for ApplicationCache
<a href="https://rdar.apple.com/143596399">rdar://143596399</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=286641">https://bugs.webkit.org/show_bug.cgi?id=286641</a>

Reviewed by Devin Rousso and Sihui Liu.

Application Cache has been deprecated for many releases, so we don&apos;t need to
keep carrying this little-used capability into the future. Rather than port this
domain to work with cross-origin iframes, let&apos;s remove it.

This domain never had any layout tests associated with it, so there are no deleted tests.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/DerivedSources-input.xcfilelist:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/inspector/protocol/ApplicationCache.json: Removed.
* Source/WebCore/Sources.txt:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/inspector/InspectorController.cpp:
(WebCore::InspectorController::createLazyAgents):
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::networkStateChangedImpl): Deleted.
(WebCore::InspectorInstrumentation::updateApplicationCacheStatusImpl): Deleted.
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::networkStateChanged): Deleted.
(WebCore::InspectorInstrumentation::updateApplicationCacheStatus): Deleted.
* Source/WebCore/inspector/agents/InspectorApplicationCacheAgent.cpp: Removed.
* Source/WebCore/inspector/agents/InspectorApplicationCacheAgent.h: Removed.
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
(WebCore::ApplicationCacheHost::notifyDOMApplicationCache):
(WebCore::ApplicationCacheHost::swapCache):
* Source/WebCore/page/Page.cpp:
(WebCore::networkStateChanged):
* Source/WebInspectorUI/UserInterface/Base/Main.js:
(WI.loaded):
* Source/WebInspectorUI/UserInterface/Controllers/ApplicationCacheManager.js: Removed.
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Models/ApplicationCacheFrame.js: Removed.
* Source/WebInspectorUI/UserInterface/Models/ApplicationCacheManifest.js: Removed.
* Source/WebInspectorUI/UserInterface/Images/TableIcons.svg:
* Source/WebInspectorUI/UserInterface/Models/ConsoleMessage.js:
* Source/WebInspectorUI/UserInterface/Models/IssueMessage.js:
* Source/WebInspectorUI/UserInterface/Models/NativeFunctionParameters.js:
* Source/WebInspectorUI/UserInterface/Protocol/ApplicationCacheObserver.js: Removed.
* Source/WebInspectorUI/UserInterface/Test.html:
* Source/WebInspectorUI/UserInterface/Test/Test.js:
(WI.loaded):
(WI.contentLoaded):
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheDetailsSidebarPanel.js: Removed.
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheFrameContentView.css: Removed.
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheFrameContentView.js: Removed.
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheFrameTreeElement.js: Removed.
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheManifestTreeElement.js: Removed.
* Source/WebInspectorUI/UserInterface/Views/ContentView.js:
(WI.ContentView.createFromRepresentedObject):
(WI.ContentView.isViewable):
* Source/WebInspectorUI/UserInterface/Views/NavigationSidebarPanel.js:
(WI.NavigationSidebarPanel.prototype._isTreeElementWithoutRepresentedObject):
* Source/WebInspectorUI/UserInterface/Views/StorageSidebarPanel.js:
(WI.StorageSidebarPanel):
(WI.StorageSidebarPanel.prototype.closed):
(WI.StorageSidebarPanel._treeSelectionDidChange):
(WI.StorageSidebarPanel.prototype._frameManifestAdded): Deleted.
(WI.StorageSidebarPanel.prototype._addFrameManifest): Deleted.
(WI.StorageSidebarPanel.prototype._frameManifestRemoved): Deleted.
(WI.StorageSidebarPanel.prototype._applicationCacheCleared): Deleted.
* Source/WebInspectorUI/UserInterface/Views/StorageTabContentView.js:
(WI.StorageTabContentView):
(WI.StorageTabContentView.isTabAllowed):
(WI.StorageTabContentView.prototype.canShowRepresentedObject):
(WI.StorageTabContentView.prototype.closed):

* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/13.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/13.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/14.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/14.5/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/15.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/15.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/16.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/16.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/12.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/12.3/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/13.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/13.3/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/Versions/Inspector-iOS-12.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-12.2.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-13.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-13.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-14.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-14.5.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-15.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-15.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-16.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-16.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.2.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-18.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-18.2.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-12.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-12.3.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-13.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-13.3.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.2.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.4.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-15.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-15.2.json:
Remove ApplicationCache domain from versioned protocol files so we don&apos;t try to set up observers and managers.

Canonical link: <a href="https://commits.webkit.org/289820@main">https://commits.webkit.org/289820@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/da2f49dc3273851944793ba0f8cbbe49e4b7a30b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87993 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7509 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42386 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92926 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38743 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90044 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7890 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15685 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67917 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/forms/constraints/infinite_backtracking.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25661 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90995 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6042 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79609 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48287 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5818 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/34017 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37850 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/80791 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76221 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34898 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94781 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/86768 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15161 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11149 "Found 3 new test failures: fast/events/event-handler-detached-document.html imported/w3c/web-platform-tests/html/capability-delegation/delegate-fullscreen-request-subframe-same-origin.https.tentative.html pdf/annotations/radio-buttons-select-second.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76773 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15416 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75465 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76012 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18710 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20410 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18820 "Found 1 new test failure: imported/w3c/web-platform-tests/preload/preload-type-match.html (failure)") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8165 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15179 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/20480 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/109262 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14921 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26275 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18366 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16703 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Base/Main.js:
```diff
@@ -1413,8 +1410,7 @@ WI.tabContentViewClassForRepresentedObject = function(representedObject)
 
     if (representedObject instanceof WI.DOMStorageObject || representedObject instanceof WI.CookieStorageObject ||
         representedObject instanceof WI.DatabaseTableObject || representedObject instanceof WI.DatabaseObject ||
-        representedObject instanceof WI.ApplicationCacheFrame || representedObject instanceof WI.IndexedDatabaseObjectStore ||
```
there's a few other values on `WI.ContentViewCookieType` that probably should be removed

burg replied:
I am handling Database in the separate patch.

dcrousso commented on Source/WebInspectorUI/UserInterface/Test/Test.js:
```diff
@@ -28,7 +28,6 @@ WI.loaded = function()
     // Register observers for events from the InspectorBackend.
     // The initialization order should match the same in Main.js.
     InspectorBackend.registerAnimationDispatcher(WI.AnimationObserver);
-    InspectorBackend.registerApplicationCacheDispatcher(WI.ApplicationCacheObserver);
```
please also remove this below
```
makeAgentGetter("ApplicationCache");
```

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/ApplicationCacheDetailsSidebarPanel.js:
```diff
@@ -1,165 +0,0 @@
-    {
-        super.initialLayout();
-
-        this._locationManifestURLRow = new WI.DetailsSectionSimpleRow(WI.UIString("Manifest URL"));
-        this._locationFrameURLRow = new WI.DetailsSectionSimpleRow(WI.UIString("Frame URL"));
```
please dont forget to update `Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js` :)

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/ApplicationCacheFrameTreeElement.js:
```diff
@@ -1,64 +0,0 @@
-
-        const title = null;
-        const subtitle = null;
-        super("application-cache-frame", title, subtitle, representedObject);
```
this has an associated icon in `Source/WebInspectorUI/UserInterface/Views/StorageIcons.css` (and `Source/WebInspectorUI/UserInterface/Images/TableIcons.svg`) that we dont need anymore

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/ApplicationCacheManifestTreeElement.js:
```diff
@@ -1,72 +0,0 @@
-    {
-        console.assert(representedObject instanceof WI.ApplicationCacheManifest);
-
-        super("application-cache-manifest", "", representedObject);
```
this has an associated icon in `Source/WebInspectorUI/UserInterface/Views/StorageIcons.css` (and `Source/WebInspectorUI/UserInterface/Images/ApplicationCacheManifest.svg`) that we dont need anymore

dcrousso commented on Source/WebInspectorUI/UserInterface/Controllers/ApplicationCacheManager.js:
```diff
@@ -1,274 +0,0 @@
-            return;
-
-        if (target.hasDomain("ApplicationCache")) {
-            target.ApplicationCacheAgent.enable();
```
please also remove this from `Source/WebInspectorUI/UserInterface/Protocol/Target.js`
```js
get ApplicationCacheAgent() { return this._agents.ApplicationCache; }
```

generally speaking, im slightly hesitant about removing this from the UI as it means we'll be dropping support when the backend technically could still work, but seeing as how this all has been disabled for so long i think it's fine

if we are gonna remove the frontend bits, however, we should manually edit all of the `Source/WebInspectorUI/UserInterface/Protocol/Legacy/*/*/InspectorBackendCommands.js` to not have the `ApplicationCache` domain as otherwise we'll still create the `WI.ApplicationCacheObserver` class, `ApplicationCacheAgent` class, etc. which is just wasteful

burg replied:
I will update the legacy protocol to remove it.

dcrousso commented on Source/JavaScriptCore/inspector/protocol/ApplicationCache.json:
```diff
@@ -1,92 +0,0 @@
-            ]
-        },
-        {
-            "name": "networkStateUpdated",
-            "parameters": [
-                { "name": "isNowOnline", "type": "boolean" }
-            ]
```
do we have any sort of replacement for this?  im guessing it's not really necessary since this has been disabled for so long anyways?

burg replied:
Nope

dcrousso reviewed (approved):
awesome! a few things to fix but overall very excited to get rid of this code :)

dcrousso commented on Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.0/InspectorBackendCommands.js:
```diff
@@ -96,7 +85,7 @@ InspectorBackend.activateDomain("Canvas", ["page"]);
 
 // Console
 InspectorBackend.registerDomain("Console", ["itml", "javascript", "page", "service-worker", "worker"]);
-InspectorBackend.registerEnum("Console.ChannelSource", {XML: "xml", JavaScript: "javascript", Network: "network", ConsoleAPI: "console-api", Storage: "storage", Appcache: "appcache", Rendering: "rendering", CSS: "css", Security: "security", ContentBlocker: "content-blocker", Media: "media", WebRTC: "webrtc", Other: "other"});
```
this should also probably be removed from `Source/JavaScriptCore/inspector/protocol/Console.json`

burg committed:
Web Inspector: remove frontend and backend support for ApplicationCache
rdar://143596399
https://bugs.webkit.org/show_bug.cgi?id=286641

Reviewed by Devin Rousso and Sihui Liu.

Application Cache has been deprecated for many releases, so we don't need to
keep carrying this little-used capability into the future. Rather than port this
domain to work with cross-origin iframes, let's remove it.

This domain never had any layout tests associated with it, so there are no deleted tests.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/DerivedSources-input.xcfilelist:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/inspector/protocol/ApplicationCache.json: Removed.
* Source/WebCore/Sources.txt:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/inspector/InspectorController.cpp:
(WebCore::InspectorController::createLazyAgents):
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::networkStateChangedImpl): Deleted.
(WebCore::InspectorInstrumentation::updateApplicationCacheStatusImpl): Deleted.
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::networkStateChanged): Deleted.
(WebCore::InspectorInstrumentation::updateApplicationCacheStatus): Deleted.
* Source/WebCore/inspector/agents/InspectorApplicationCacheAgent.cpp: Removed.
* Source/WebCore/inspector/agents/InspectorApplicationCacheAgent.h: Removed.
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
(WebCore::ApplicationCacheHost::notifyDOMApplicationCache):
(WebCore::ApplicationCacheHost::swapCache):
* Source/WebCore/page/Page.cpp:
(WebCore::networkStateChanged):
* Source/WebInspectorUI/UserInterface/Base/Main.js:
(WI.loaded):
* Source/WebInspectorUI/UserInterface/Controllers/ApplicationCacheManager.js: Removed.
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Models/ApplicationCacheFrame.js: Removed.
* Source/WebInspectorUI/UserInterface/Models/ApplicationCacheManifest.js: Removed.
* Source/WebInspectorUI/UserInterface/Images/TableIcons.svg:
* Source/WebInspectorUI/UserInterface/Models/ConsoleMessage.js:
* Source/WebInspectorUI/UserInterface/Models/IssueMessage.js:
* Source/WebInspectorUI/UserInterface/Models/NativeFunctionParameters.js:
* Source/WebInspectorUI/UserInterface/Protocol/ApplicationCacheObserver.js: Removed.
* Source/WebInspectorUI/UserInterface/Test.html:
* Source/WebInspectorUI/UserInterface/Test/Test.js:
(WI.loaded):
(WI.contentLoaded):
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheDetailsSidebarPanel.js: Removed.
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheFrameContentView.css: Removed.
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheFrameContentView.js: Removed.
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheFrameTreeElement.js: Removed.
* Source/WebInspectorUI/UserInterface/Views/ApplicationCacheManifestTreeElement.js: Removed.
* Source/WebInspectorUI/UserInterface/Views/ContentView.js:
(WI.ContentView.createFromRepresentedObject):
(WI.ContentView.isViewable):
* Source/WebInspectorUI/UserInterface/Views/NavigationSidebarPanel.js:
(WI.NavigationSidebarPanel.prototype._isTreeElementWithoutRepresentedObject):
* Source/WebInspectorUI/UserInterface/Views/StorageSidebarPanel.js:
(WI.StorageSidebarPanel):
(WI.StorageSidebarPanel.prototype.closed):
(WI.StorageSidebarPanel._treeSelectionDidChange):
(WI.StorageSidebarPanel.prototype._frameManifestAdded): Deleted.
(WI.StorageSidebarPanel.prototype._addFrameManifest): Deleted.
(WI.StorageSidebarPanel.prototype._frameManifestRemoved): Deleted.
(WI.StorageSidebarPanel.prototype._applicationCacheCleared): Deleted.
* Source/WebInspectorUI/UserInterface/Views/StorageTabContentView.js:
(WI.StorageTabContentView):
(WI.StorageTabContentView.isTabAllowed):
(WI.StorageTabContentView.prototype.canShowRepresentedObject):
(WI.StorageTabContentView.prototype.closed):

* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/13.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/13.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/14.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/14.5/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/15.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/15.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/16.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/16.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/17.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/18.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/12.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/12.3/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/13.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/13.3/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/14.4/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.0/InspectorBackendCommands.js:
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/15.2/InspectorBackendCommands.js:
* Source/WebInspectorUI/Versions/Inspector-iOS-12.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-12.2.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-13.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-13.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-14.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-14.5.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-15.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-15.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-16.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-16.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.2.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-17.4.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-18.0.json:
* Source/WebInspectorUI/Versions/Inspector-iOS-18.2.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-12.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-12.3.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-13.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-13.3.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.2.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-14.4.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-15.0.json:
* Source/WebInspectorUI/Versions/Inspector-macOS-15.2.json:
Remove ApplicationCache domain from versioned protocol files so we don't try to set up observers and managers.

Canonical link: https://commits.webkit.org/289820@main
